src/main/scala/query/MatchCreateQuery.scala

```diff
@@ -0,0 +1,74 @@
+package com.originate.scalypher
+
+import action.ReturnAction
+import action.ReturnAll
+import action.ReturnDistinct
+import action.ReturnReference
+import path.AnyNode
+import path.AnyRelationship
+import path.Node
+import path.Path
+import path.Relationship
+import where.Reference
+import types._
+
+trait MatchCreateQuery extends Query {
+
+  def returnAction: Option[ReturnAction]
+
+  def returns(reference: Reference, rest: Reference*): MatchCreateQuery =
+    withReturnAction(ReturnReference(reference, rest: _*))
+
+  def returnDistinct(reference: Reference, rest: Reference*): MatchCreateQuery =
+    withReturnAction(ReturnDistinct(reference, rest: _*))
+
+  def returnAll: MatchCreateQuery =
+    withReturnAction(ReturnAll)
+
+  def getReturnColumns: Set[String] =
+    returnAction match {
+      case Some(action) => matchActionToReturnColumns(action)
+      case _ => Set.empty
+    }
+
+  protected def withReturnAction(action: ReturnAction): MatchCreateQuery
+
+  protected def forcedCreateReferenceables: Set[Referenceable] = Set.empty
+
+  protected def modifiedReferenceableMap: ReferenceableMap = {
+    val returnReferenceables = returnAction map (_.referenceables) getOrElse Set.empty
+    val returnMap = referenceableMap filterKeys { referenceable =>
+      (returnReferenceables contains referenceable) || (forcedCreateReferenceables contains referenceable)
```

Comment:
i gave a hook/abstract member to add in other referenceables aside from those coming from the `returnReferenceables` (`forecedCreateReferenceables`, specifically used since you can set properties on things in a merge query)... still working on a better name for this...